certbot/tests/error_handler_test.py

```diff
@@ -1,10 +1,37 @@
 """Tests for certbot.error_handler."""
+import contextlib
+import os
 import signal
 import sys
 import unittest
 
 import mock
 
+def get_signals(signums):
+    """Get the handlers for an iterable of signums."""
+    return dict((s, signal.getsignal(s)) for s in signums)
+
+
+def set_signals(sig_handler_dict):
+    """Set the signal (keys) with the handler (values) from the input dict."""
+    tuple(signal.signal(s, h) for (s, h) in sig_handler_dict.items())
+
+
+@contextlib.contextmanager
+def signal_receiver(signums):
+    """Context manager to catch signals"""
+    signals = []
+    prev_handlers = {}
+    prev_handlers = get_signals(signums)
+    set_signals(dict((s, lambda s, _: signals.append(s)) for s in signums))
+    yield signals
+    set_signals(dict((s, prev_handlers[s]) for s in signums))
```

Comment:
We should be able to just pass in `prev_handlers` to `set_signals` instead of building a new dict.